test/tests/functional/pbs_holidays.py

```diff
@@ -0,0 +1,148 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import datetime
+from tests.functional import *
+
+
+class TestHolidays(TestFunctional):
+
+    """
+    This test suite tests if PBS scheduler's holidays file feature
+    works correctly
+    """
+    days = ["monday", "tuesday", "wednesday", "thursday", "friday",
+            "saturday", "sunday"]
+    np_queue = "np_workq"
+    p_queue = "p_workq"
+    cur_year = datetime.datetime.today().year
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        # Create a non-prime queue
+        a = {ATTR_enable: True, ATTR_start: True, ATTR_qtype: "Execution"}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id=self.np_queue)
+
+        # Create a prime queue
+        a = {ATTR_enable: True, ATTR_start: True, ATTR_qtype: "Execution"}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id=self.p_queue)
+
+    def check_prime(self):
+        """
+        Helper function to test that it is prime-time right now
+        """
+        # Submit a job to the non-prime queue
+        a = {ATTR_queue: self.np_queue}
+        j1 = Job(attrs=a)
+        jid1 = self.server.submit(j1)
+
+        # Verify that the job does not run as it is 24hrs prime-time
+        self.server.expect(JOB, {'job_state': 'Q'}, jid1)
```

Comment:
Check to see that comment is set before checking for job_state: Q.  There is a race that you do your expect() before the scheduler even sees the job.